forum/communications/consumers.py

```diff
@@ -66,3 +72,49 @@ async def chat_message(self, event):
         logger.info(f"Received message: {message}")
 
         await self.send(text_data=json.dumps({"message": message}))
+
+
+class NotificationConsumer(AsyncWebsocketConsumer):
+    async def connect(self):
+        self.user_id = self.scope['url_route']['kwargs']['user_id']
+        self.users_group_name = f'{self.user_id}'
+
+        try:
+            # Join the user notification group
+            await self.channel_layer.group_add(
+                self.users_group_name,
+                self.channel_name
+            )
+            logger.info(f"Client connected to notification group {self.users_group_name}")
+            await self.accept()
+
+        except Exception as e:
+            logger.error(f"Error connecting to notification group {self.users_group_name}: {e}")
+
+    async def disconnect(self, close_code):
+        logger.info(f"Disconnected from notification group  {self.users_group_name}")
+        await self.channel_layer.group_discard(
+            self.users_group_name,
+            self.channel_name
+        )
+
+    async def receive(self, text_data):
+        try:
+            text_data_json = json.loads(text_data)
+            notification = text_data_json["message"]
+
+            # Send message to room group
+            await self.channel_layer.group_send(
+                self.users_group_name, {"type": "notification.message", "message": notification}
+            )
+
+        except json.JSONDecodeError as e:
+            logger.error(f"Invalid JSON data received: {text_data}, error: {e}")
+            print(f"Error decoding JSON: {e}")
+
+    # Receive notification from notification group
+    async def notification_message(self, event):
+        notification = event["notification"]
+        logger.info(f"Received notification: {notification}")
+
+        await self.send(text_data=json.dumps({"message": notification}))
```

Comment:
If notifications are malformed or incorrectly sent, the consumer only logs the error without fallback behavior.
Add fallback handling or retries to avoid message loss.

Reply:
Added max retry and global exception handler